lib/openxml/docx/properties/table_cell_margin.rb

```diff
@@ -0,0 +1,32 @@
+module OpenXml
+  module Docx
+    module Properties
+      class TableCellMargin < WidthProperty
+        attr_accessor :tag
+
+        attribute :type, expects: :valid_width_type
+        attribute :width, expects: :positive_integer, displays_as: :w
+
+        def initialize(tag)
+          @tag = tag
+          raise ArgumentError, invalid_tag_message unless valid_tag? tag
+        end
+
+      private
+
+        def valid_tag?(value)
+          ok_tags.include? value
+        end
+
+        def invalid_tag_message
+          "Invalid value for #{name}; acceptable values are #{ok_tags.join(", ")} (provided: #{tag.inspect})"
+        end
+
+        def ok_tags
+          %i(top start bottom end)
+        end
```

Comment:
You've used lines 12–27 in several places. Can it be extracted?

Maybe on the Property base class?

```ruby
class Property
  self << class
    attr_reader :allowed_tags

    def tag!(tags)
      attr_accessor :tag
      @allowed_tags = tags
    end
  end

  def initialize(tag, *args)
    return unless self.class.allowed_tags
    unless self.class.allowed_tags.include?(tag)
      raise ArgumentError, "Invalid tag name for #{name}: #{tag.inspect}. It should be one of #{self.class.allowed_tags.join(", ")}."
    end
    @tag = tag
  end
end
```


```ruby
module OpenXml
  module Docx
    module Properties
      class TableCellMargin < WidthProperty
        tag! %i(top start bottom end)

        attribute :type, expects: :valid_width_type
        attribute :width, expects: :positive_integer, displays_as: :w

      end
    end
  end
end
```

This way `tag!` looks more like Gene's `tag` statement in the property DSL.

But maybe `tag_may_be` or `tag_is_one_of` or `tag_is_any` would be better than the unexplained `!` on `tag!`.

Maybe we even modify `tag` so that you say:

```ruby
module OpenXml
  module Docx
    module Properties
      class TableCellMargin < WidthProperty
        tag one_of: %i(top start bottom end)
        # ...
```